metals/src/main/scala/scala/meta/internal/metals/Messages.scala

```diff
@@ -370,6 +370,38 @@ object Messages {
     }
   }
 
+  object ChooseBspServer {
+    case class Request(
+        params: ShowMessageRequestParams,
+        mapping: Map[String, String]
+    )
+
+    def request(
```

Comment:
So this is very similiar to the request in `SelectBspServer` down below. The one down below however is only dealing with connection details, and it accounts for things like,  two different entries for the same build server that have the same name, but different version. I honestly don't think this would ever happen, but also didn't want to remove it since it is used in another place (only in the scenario when you open in a non Bloop workspace for the first time and 2 bsp entries are found). Since in this case we aren't guaranteed to have the connection details because the entry doesn't yet exist, we don't do that extra check. Do you think this is ok?

Reply:
I don't think keeping the logic below is needed, I would instead warn if there are multiple servers with the same version and just suggest the newest. As you said, this scenario has never happened as each tool always replaces it's config and it would require some manual shenanigans for this to actually pop up.

Reply:
Sounds good, I got rid of this and added some logic that if there are two of the same name the newest is chose and there is a warning logged.